src/webapp/components/profile-modal/ProfileModal.tsx

```diff
@@ -0,0 +1,89 @@
+import React from "react";
+import { Modal, Avatar, CardContent, Card } from "@material-ui/core";
+import styled from "styled-components";
+
+import i18n from "../../../utils/i18n";
+import { Button } from "../button/Button";
+
+type ProfileModalProps = {
+    name: string;
+    children: React.ReactNode;
+    avatarSize?: "small" | "medium";
+    alt?: string;
+    src?: string;
+    open: boolean;
+    onClose: () => void;
+};
+
+export const ProfileModal: React.FC<ProfileModalProps> = React.memo(
+    ({ children, src, alt, open = false, onClose, name }) => {
+        return (
+            <Modal
+                aria-labelledby={`modal-${name}`}
+                aria-describedby={`${name}-profile`}
+                open={open}
+                onClose={onClose}
+                hideBackdrop
+            >
+                <StyledCard variant="outlined">
+                    <Name>{name}</Name>
+                    <Content>
+                        <AvatarContainer>
+                            <Avatar alt={alt} src={src} />
+                        </AvatarContainer>
+                        <StyledCardContent>{children}</StyledCardContent>
+                    </Content>
+                    <Footer>
+                        <Button onClick={onClose}>{i18n.t("Close")}</Button>
+                    </Footer>
+                </StyledCard>
```

Comment:
[subjective] Components at the same level (all the more when multi-line) are somewhat more readable when separated by a blank line. It's similar to what we do in normal code.